File: 0_liste&dict/Py/lc3375.py
from typing import List


class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        s = set()
        for x in nums:
            if x < k:
                return -1
            elif x > k:
                s.add(x)
        return len(s)

# Time complexity: O(n)
# Space complexity: O(n) (the set)

File: 0_liste&dict/C++/lc3375.cpp
class Solution {
    public:
        int minOperations(vector<int>& nums, int k) {
            unordered_set<int> s;
            for (int x : nums){
                if (x < k){
                    return -1;
                }else if (x > k){
                    s.insert(x);
                }
            }
            return s.size();
        }
    };